vaadin-date-time-picker-flow-demo/src/main/java/com/vaadin/flow/component/datetimepicker/demo/DateTimePickerView.java

```diff
@@ -0,0 +1,117 @@
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Locale;
+
+import com.vaadin.flow.component.datetimepicker.DateTimePicker;
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.Label;
+import com.vaadin.flow.component.html.NativeButton;
+import com.vaadin.flow.demo.DemoView;
+import com.vaadin.flow.router.Route;
+
+/**
+ * View for {@link DateTimePicker} demo.
+ *
+ * @author Vaadin Ltd
+ */
+@Route("vaadin-date-time-picker")
+public class DateTimePickerView extends DemoView {
+
+    @Override
+    public void initView() {
+        createSimpleDateTimePicker();
+        createLocaleChangeDateTimePicker();
+        addCard("Additional code used in the demo",
+                new Label("These methods are used in the demo."));
+    }
+
+    private void createSimpleDateTimePicker() {
+        Div message = createMessageDiv("simple-picker-message");
+
+        // begin-source-example
+        // source-example-heading: Simple date time picker
+        DateTimePicker dateTimePicker = new DateTimePicker();
+
+        dateTimePicker.addValueChangeListener(
+                event -> updateMessage(message, dateTimePicker));
+        // end-source-example
+
+        dateTimePicker.setId("simple-picker");
+        addCard("Simple date time picker", dateTimePicker, message);
+    }
+
+    private void createLocaleChangeDateTimePicker() {
+        Div message = createMessageDiv("Customize-locale-picker-message");
+        // begin-source-example
+        // source-example-heading: Date picker with customize locales
+        // By default, the dateTimePicker uses the current UI locale
+        DateTimePicker dateTimePicker = new DateTimePicker();
+        NativeButton locale1 = new NativeButton("Locale: US");
+        NativeButton locale2 = new NativeButton("Locale: UK");
+        NativeButton locale3 = new NativeButton("Locale: CHINA");
+
+        locale1.addClickListener(e -> {
+            dateTimePicker.setLocale(Locale.US);
+            updateMessage(message, dateTimePicker);
+        });
+        locale2.addClickListener(e -> {
+            dateTimePicker.setLocale(Locale.UK);
+            updateMessage(message, dateTimePicker);
+        });
+        locale3.addClickListener(e -> {
+            dateTimePicker.setLocale(Locale.CHINA);
+            updateMessage(message, dateTimePicker);
+        });
+
+        dateTimePicker.addValueChangeListener(
+                event -> updateMessage(message, dateTimePicker));
+        // end-source-example
+        locale1.setId("Locale-US");
+        locale2.setId("Locale-UK");
+        dateTimePicker.setId("locale-change-picker");
+        addCard("Date picker with customize locales", dateTimePicker, locale1,
+                locale2, locale3, message);
+    }
+
+    // begin-source-example
+    // source-example-heading: Additional code used in the demo
+    /**
+     * Additional code used in the demo
+     */
```

Comment:
This comment can be removed, as it just duplicates the header text right above it.

Reply:
Done.